modules/storages/app/components/storages/admin/new_storage_button_component.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2023 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+#
+module Storages::Admin
+  class NewStorageButtonComponent < ApplicationComponent
+    options scheme: :primary,
+            size: :medium,
+            tag: :a
+
+    def call
+      render(Primer::Beta::Button.new(**button_options)) do |button|
+        button.with_leading_visual_icon(icon: :plus)
+        label
+      end
+    end
+
+    private
+
+    def button_options
+      { scheme:,
+        size:, tag:,
+        href: new_admin_settings_storage_path,
+        aria: { label: },
+        title: label }
```

Comment:
This will not be helpful from a screen reader perspective. The text being read with Orca is "Storage (pause) visited link".
[The title element should always be avoided](https://www.a11yproject.com/posts/title-attributes/), except for iframe tag. It is not accessible.
The aria-label is useless if it is the same as the text of the element.

To improve it from an accessibility stand point:
* if it acts as a button, use `role="button"`
* the `+` icon is invisible to the user. A sighted user might understand that it is a metaphor for "Add", but blind people will not. They will only "see" a "Storage" button without knowing the action
  * You can add `aria-label="Add new storage"`
  * Changing the label from "Storage" to "Add new storage" would maybe make it clearer for some sighted users. That's something to discuss with the UX people.

With these modifications, Orca now reads it as "Add new storage (pause) push button".

Reply:
This is super helpful Christophe, thank you 👍🏾